src/runtime/classbase.cs

```diff
@@ -18,18 +18,21 @@ namespace Python.Runtime
     [Serializable]
     internal class ClassBase : ManagedType
     {
+        [NonSerialized]
+        internal List<string> dotNetMembers;
         internal Indexer indexer;
-        internal Type type;
+        internal MaybeType type;
 
         internal ClassBase(Type tp)
         {
+            dotNetMembers = new List<string>();
             indexer = null;
             type = tp;
         }
 
         internal virtual bool CanSubclass()
         {
-            return !type.IsEnum;
+            return !type.Value.IsEnum;
```

Comment:
Should this return false or throw on an invalid type?

Reply:
I think it should throw. It would be false to say that something that doesn't exist can't subclass; we can't know since it doesn't exist.
Also, the code that calls `CanSubclass` throws another kind of exception when it's false `Exceptions.RaiseTypeError("delegates, enums and array types cannot be subclassed");`